sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(System.Net.ClientModel.ModelReaderWriterFormat left, System.Net.ClientModel.ModelReaderWriterFormat right) { throw null; }
+        public static implicit operator System.Net.ClientModel.ModelReaderWriterFormat (string value) { throw null; }
+        public static bool operator !=(System.Net.ClientModel.ModelReaderWriterFormat left, System.Net.ClientModel.ModelReaderWriterFormat right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class ModelReaderWriterOptions
+    {
+        public ModelReaderWriterOptions() { }
+        public ModelReaderWriterOptions(System.Net.ClientModel.ModelReaderWriterFormat format) { }
+        public static System.Net.ClientModel.ModelReaderWriterOptions DefaultWireOptions { get { throw null; } }
+        public System.Net.ClientModel.ModelReaderWriterFormat Format { get { throw null; } }
+    }
+    public partial class NullableResult<T> : System.Net.ClientModel.Result
+    {
+        internal NullableResult() { }
+        public virtual bool HasValue { get { throw null; } }
+        public virtual T? Value { get { throw null; } }
+        public override System.Net.ClientModel.Core.MessageResponse GetRawResponse() { throw null; }
+    }
+    public partial class RequestOptions : System.Net.ClientModel.Core.PipelineOptions
+    {
+        public RequestOptions() { }
+        public virtual System.Threading.CancellationToken CancellationToken { get { throw null; } set { } }
+        public static System.Threading.CancellationToken DefaultCancellationToken { get { throw null; } set { } }
+        public virtual System.Net.ClientModel.Core.ErrorBehavior ErrorBehavior { get { throw null; } set { } }
+        public virtual void Apply(System.Net.ClientModel.Core.ClientMessage message) { }
+    }
+    public abstract partial class Result
+    {
+        protected Result() { }
+        public static System.Net.ClientModel.NullableResult<T> FromNullableValue<T>(T? value, System.Net.ClientModel.Core.MessageResponse response) { throw null; }
+        public static System.Net.ClientModel.Result FromResponse(System.Net.ClientModel.Core.MessageResponse response) { throw null; }
+        public static System.Net.ClientModel.Result<T> FromValue<T>(T value, System.Net.ClientModel.Core.MessageResponse response) { throw null; }
+        public abstract System.Net.ClientModel.Core.MessageResponse GetRawResponse();
+    }
+    public partial class Result<T> : System.Net.ClientModel.NullableResult<T>
+    {
+        internal Result() { }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool HasValue { get { throw null; } }
+        public override T Value { get { throw null; } }
+    }
+    public partial class UnsuccessfulRequestException : System.Exception
```

Comment:
naming: RequestErrorException? Seems more consistent with concepts like [IsError](https://github.com/Azure/azure-sdk-for-net/pull/39135/files#diff-e65b266db2070f484891063291be320185806047f938e84579c13f4366e7c5a5R125)

Reply:
RequestErrorException is not a bad name.

Reply:
I think we had this name in this version: https://github.com/Azure/azure-sdk-for-net/blob/9fdd76668e7c595b8358649496e61fb6b2577990/sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.netstandard2.0.cs#L25, but a concern was named that we needed a verb to describe what happened to the request.  @KrzysztofCwalina, do you still have this concern?

We also looked at `MessageFailedException` which added the verb `Failed` into it.
For consistency with BCL, they have an exception called [HttpRequestException](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httprequestexception?view=net-7.0), so we could also just called it `MessageRequestException` or `PipelineRequestException` if we wanted to align with that.

I spent some time looking at names of exceptions derived from [Exception](https://learn.microsoft.com/en-us/dotnet/api/system.exception?view=net-7.0) - there is a precedent for saying "<Bad><Noun>Exception", i.e. UnsupportedPolicyOptionsException, UntrustedRecipientException, AmbiguousImplementationException etc.  In this pattern we could do "FailedRequestException" but that would leave us with ClientModel.FailedRequestException and Azure.RequestFailedException, which I think we mostly agree is not a good thing :)